sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -1007,6 +1007,8 @@ def test_issue_10122():
     assert solve(abs(x) + abs(x - 1) - 1 > 0, x
         ) == Or(And(-oo < x, x < S.Zero), And(S.One < x, x < oo))
 
+def test_issue_20755():
+    assert Piecewise((x, tan(x) >= 0)).integrate() == Piecewise((S.NaN, x <= 0), (x**2/2, x < pi/2), (S.NaN, True))
```

Comment:
The condition in this case is periodic. The main part around `x=0` should be `-pi/2 < x < pi/2` but there are also infinitely many other cases like `pi/2 < x < 3*pi/2`.

I'm not sure what the best way to represent this is. Does this work correctly for definite integrals?

Reply:
Oh of course it should be `0 <= x < pi/2` because it's `tan(x) >= 0`. Here it is `0 < tan(x) < pi/2` though so this would give `NaN` at `x=0`.

Reply:
Isn't `tan(x) >= 0` True at `x = 0`. Hence it should be equal to `x` i.e. `0`. Is there a specific reason why there should be `NaN` at that point ?

Reply:
> Does this work correctly for definite integrals?

It should work since the change being made here only deals with changes in the condition as `<=` or `<`. And If that equation is not defined at that point which by default means the equation shouldn't be defined there and the condition be set to `<` even if in case it was `<=` before.

Reply:
I mean like:
```python
In [1]: p = Piecewise((x, tan(x) >= 0))

In [2]: integrate(p, (x, pi, pi+1))
Out[2]: nan
```

Reply:
I see. But this seems like a whole other issue altogether. The cause of this is probably because integrate evaluates the case for `tan(x)` only between `0` to `pi` even tho it's periodic. Probably since there are infinite such piecewise functions. 

Something like `integrate(p, (x, 0, pi/2))` works as expected.